https://chromiumcodereview.appspot.com/9443040

fix up build warning:

In file included from .../base/lazy_instance.h:45:0,
                 from ../shill/proxy_factory.h:11,
                 from cellular_capability.cc:13:
.../base/threading/thread_restrictions.h:89:15: error: unused parameter 'allowed' [-Werror=unused-parameter]
.../base/threading/thread_restrictions.h:91:15: error: unused parameter 'allowed' [-Werror=unused-parameter]
cc1plus: all warnings being treated as errors

--- a/base/threading/thread_restrictions.h
+++ b/base/threading/thread_restrictions.h
@@ -8,6 +8,8 @@
 #include "base/base_api.h"
 #include "base/basictypes.h"
 
+#include "base/compiler_specific.h"
+
 namespace base {
 
 // Certain behavior is disallowed on certain threads.  ThreadRestrictions helps
@@ -86,9 +88,9 @@
 #else
   // In Release builds, inline the empty definitions of these functions so
   // that they can be compiled out.
-  static bool SetIOAllowed(bool allowed) { return true; }
+  static bool SetIOAllowed(ALLOW_UNUSED bool allowed) { return true; }
   static void AssertIOAllowed() {}
-  static bool SetSingletonAllowed(bool allowed) { return true; }
+  static bool SetSingletonAllowed(ALLOW_UNUSED bool allowed) { return true; }
   static void AssertSingletonAllowed() {}
 #endif
 
